app/services/foreman_bootdisk/renderer.rb

```diff
@@ -4,15 +4,30 @@ module ForemanBootdisk
   class Renderer
     include ::Foreman::Renderer
     include Rails.application.routes.url_helpers
+    delegate :logger, :to => :ForemanBootdisk
 
-    def generic_template_render
+    def generic_template_render(subnet=nil)
       tmpl = ProvisioningTemplate.find_by_name(Setting[:bootdisk_generic_host_template]) || raise(::Foreman::Exception.new(N_('Unable to find template specified by %s setting'), 'bootdisk_generic_host_template'))
-      @host = Struct.new(:token, :subnet).new(nil, nil)
-      unattended_render(tmpl.template)
+
+      if subnet.present?
+        # rendering a subnet-level bootdisk requires tricking the renderer into thinking it has a
+        # valid host, with a token, and with a tftp proxy
+        @host = Struct.new(:token, :subnet).new(
+          Struct.new(:value).new('faketoken'),
+          subnet
+        )
+      else
+        @host = Struct.new(:token, :subnet).new(nil,nil)
+      end
+      rtmpl = unattended_render(tmpl.template)
+      # remove the token from subnet-level generation, since this is meant to be generic
+      rtmpl.gsub!(/(?<=iPXE\?)token=faketoken\&(?=mac=)/,'')
```

Comment:
I am going to remove this `gsub!` because I am unable to generate host-based generic image with a token. I simply don't see them here, therefore this is useless?